drafts/draft-0005.md

```diff
@@ -0,0 +1,138 @@
+## Preamble
+
+```
+SEP: <to be assigned>
+Title: Data Entry Namespaces
+Author: Mister.Ticot <mister.ticot@cosmic.plus>
+Status: Draft
+Created: 2018-10-26
+```
+
+## Simple Summary
+
+Defines a standard syntax to represent namespaces in data entries.
+
+## Abstract
+
+This SEP provides a standard way to represent
+[namespaces](https://en.wikipedia.org/wiki/Namespace) into Stellar accounts
+data entries, and provides guideline about how to parse them and to handle
+edge cases.
+
+## Motivation
+
+As a generic mechanism to store key/value pairs on the ledger, data entries
+accept a very wide range of use cases. If each actor use it following its own
+syntax, it will lead to a confusing experience.
+
+Introducing namespaces allow to hierarchize and organize data, in such a way 
+that sorting the key/value entries will result in a readable output even when 
+using them for various purpose on the same account.
+
+It also permit to parse the whole tree at once, and provides guidelines for a 
+generic API that would allow direct access to data entries, by converting data 
+tree into JSON for example. This could make sense for some sensitive data that 
```

Comment:
Can you provide an example as to what this means?

Reply:
I assume you asked the same question twice ? This is not clear to me which part you're pointing to.

**Avoiding name conflicts:** This is generally why we use namespaces.

**Tidy looking:** This is not only about having a tidy looking object, this is also about displaying a clear and understandable data entries tree to the user. It's more professional than:

```
anchorRedirect = true
ticot.multisig.protocol = ...
ticot.multisig.options = ...
FAVORITE_COIN = ...
my_eth_entry_point = ...
pzY9c8 = ...
SuperAnchor:BTCwallet = ...
tradeAccount = ...
tradeOpts = ...
```

**Consistency:** It is clear that if each SEP use different syntax to name data entries it's quickly going to be a mess.

**Promoting standard solutions:** Defining a few standard fields and their usage would help there (like wallet,config,profile). Could be done with dedicated SEP. It allows dev to know exactly were to look for data rather than jungling with each vendor-specific solution.

**Parsing the whole tree at once:** All data entries respecting this SEP can be parsed at once with a tiny dedicated util. This will make working with data entries dev-friendly. Example, if you want a list of user external wallets:

```js
function externalWallets (account) {
  return SDK.AccountResponse.dataTree(account).wallet
}
```

**Turning into JSON:** This is only a side effect of this SEP, so maybe I shouldn't have mentioned it but it is clearly an interesting use case. It makes easy to serve data as nicely formatted JSON directly from the blockchain, without having the client needing to use a SDK or even knowing about it. The use case here are very broad so it's mainly a matter of imagination.

**Example:**
* I want to have my federation server and assets configuration locked under my 5-signers account and needs at least 60% votes to be changes (provided you read the data from a GET request)
* I want to serve live the data of my daily audits, that have to be signed by at least 2 different independent parties.
* I want to publish the shasum of software release, signed by 3 contributors at least
* ... and so on ...

Once a trusted server (Horizon?) offers data account entries as well-formatted JSON responses, it's only a matter of signing a few transaction and your data is served.

Reply:
This particular comment I was more wondering about how it improves things for the sensitive data case you mentioned.

Reply:
I'll also notes that since SEPs are mostly just ecosystem enhancements and not really baked into the core that there's nothing stopping people from polluting/using inconsistent data entries. I really think this is application/use case specific -- the main benefit I can see would be "reserving" some namespaces for specific use-cases, but that would really be the responsibility of other SEPs to define those.

Reply:
This is not meant to prevent free use of data entries, this is meant to provide guidelines for who want to use them a structured, compatible & consistent way; which is especially desirable when it comes to writing SEP:

This is not a revolutionary SEP, this is only a tiny proposal based on the fact that naming things with consistency lead to a clearer environment than multiplying syntaxes. This really is one of the basis of good programming.